### PR TITLE
Add animal-sniffer-maven-plugin to check JVM compatibility

### PR DESCRIPTION
**Pull request by vbauer**

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@vbauer__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <span>You've already signed the contribution license agreement. Thanks!</span>
        <span>We will now validate the agreement and then real humans will evaluate your PR.</span>

TTYL, AZPRBOT;

**Comment by serious6** [comment]
Hi @vbauer can you provide a short introducion of that plugin and the resulting benefit?

**Comment by vbauer** [comment]
@serious6 Hi, yes, of course. This plugin makes very simple, but useful thing - it checks that all used in project classes/methods are contained in the specified JVM's run-time library. maven-compiler-plugin can't do it (it just checks a source level).

It means that you can build project using Java 6, Java 7, Java 8, etc. and do not care that you can accidentally use some classes or methods that are available only in your JDK (for example: JDK 1.8). You can check compatibility with JDK 6 on your local machine (instead of checking it on Travis CI or installing JDK 1.6). Everything works in the auto-mode, so it will allow to minimize human-related errors.

File: pom.xml [hunk 1]
```diff
@@ -235,6 +235,27 @@
...
             </plugin>
+            <plugin>
+                <groupId>org.codehaus.mojo</groupId>
+                <artifactId>animal-sniffer-maven-plugin</artifactId>
+                <version>1.14</version>
+                <configuration>
+                    <signature>
+                        <groupId>org.codehaus.mojo.signature</groupId>
+                        <artifactId>java16-sun</artifactId>
+                        <version>1.0</version>
```
Comment (serious6) [inline pom.xml]: Latest version might be [1.1](http://search.maven.org/#browse%7C-379312183). Or is there a reason you used version 1.0 here?

**Comment by serious6** [comment]
Thanks this really makes sense. I was wondering if we maybe would like to have this with the travis build only since a failing build would lead to a discussion whether a restrictive test lifecycle wold maybe prevent contributors from using newer java functions from the beginning. so maybe there will be changes that are worth upgrading to a newer java version in the future.

**Comment by vbauer** [comment]
Should I provide some additional information or something else? (ex: move plugin in separate profile)

**Comment by serious6** [comment]
Hmm I am still wondering what benefit we will have with this. Since travis runs the build against all available jdks. The only thing I see is that we can skip building with multiple jdks. In this case we should answer the question which one will be more stable.

**Comment by vbauer** [comment]
It is possible to:
- use new versions of JDK on your local machine (instead of installing JDK 6)
- check compatibility with JDK 6 on local machine instead of sending PR and checking it on Travis CI (with the following fixes and sending new PR, again and again..)

**Comment by serious6** [comment]
@vbauer ok thanks for the contribution and for mentioning this plugin.

**Pull request merged**